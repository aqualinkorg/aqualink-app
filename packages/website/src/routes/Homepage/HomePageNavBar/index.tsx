import React from "react";
import { AppBar, Toolbar, Grid, Button } from "@material-ui/core";

const HomePageNavBar = () => {
  return (
    <AppBar position="static">
      <Toolbar>
        <Grid item xs={12}>
          <Grid container justify="flex-end" alignItems="center">
            <Grid item>
              <Button variant="contained" color="secondary">
                Log In
              </Button>
            </Grid>
          </Grid>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};

export default HomePageNavBar;
