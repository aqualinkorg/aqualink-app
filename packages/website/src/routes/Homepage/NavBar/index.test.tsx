import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";

import HomePageNavBar from ".";

const mockStore = configureStore([]);

describe("NavBar", () => {
  let element: HTMLElement;
  beforeEach(() => {
    const store = mockStore({
      user: {
        userInfo: null,
        loading: false,
        error: null,
      },
    });

    store.dispatch = jest.fn();

    element = render(
      <Provider store={store}>
        <HomePageNavBar />
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(element).toMatchSnapshot();
  });
});
