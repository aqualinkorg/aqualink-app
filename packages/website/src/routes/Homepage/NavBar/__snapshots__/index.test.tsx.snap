// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render with given state from Redux store 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary HomepageNavBar-appBar-1 MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-5"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            Aqua
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h4"
            style="color: rgb(138, 198, 222);"
          >
            link
          </h4>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-6"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                SIGN IN
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                REGISTER
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
